src/DotNetWorker.Grpc/Http/GrpcHttpRequestData.cs

```diff
@@ -20,11 +20,48 @@ internal class GrpcHttpRequestData : HttpRequestData, IAsyncDisposable, IDisposa
         private HttpHeadersCollection? _headers;
         private Stream? _bodyStream;
         private bool _disposed;
+        private readonly Lazy<IReadOnlyCollection<IHttpCookie>> _cookies;
 
         public GrpcHttpRequestData(RpcHttp httpData, FunctionContext functionContext)
             : base(functionContext)
         {
             _httpData = httpData ?? throw new ArgumentNullException(nameof(httpData));
+            _cookies = new Lazy<IReadOnlyCollection<IHttpCookie>>(() =>
+            {
+                if(_headers is null)
+                {
+                    return Array.Empty<IHttpCookie>();
+                }
+
+                var cookieString = _headers.FirstOrDefault(item => item.Key == "Cookie").Value;
+
+                if (cookieString != null && cookieString.Count() > 0)
+                {
+
+                    return ToHttpCookies(cookieString);
+                }
+
+                return Array.Empty<IHttpCookie>();
+
+
+            });
+        }
+
+        private IReadOnlyCollection<IHttpCookie> ToHttpCookies(IEnumerable<string> cookieString)
```

Comment:
nit: arg name sounds singular while the type is a collection. May be you can change the type to string and pass the first item in line 41 ?